### DISCUSSION TITLE
Feat: Connection quality

### DISCUSSION THREAD

**Discussion by jibon57** [8xFF/atm0s-media-server#389] [Ideas]
Based on different criteria server can send quality of user's connection e.g. excellent, good, bad etc...

**Comment by jibon57** [comment]
For this feature we can use server event: https://github.com/8xFF/atm0s-media-server/blob/faf69e25f18461606cc965229863a30e6ac0edbc/packages/protocol/proto/sdk/session.proto#L242 to make sure all the user got the status of every users' connection status. This way from application we can easily get data for all connected users. Related: https://github.com/8xFF/atm0s-media-server/issues/387

**Reply by giangndm** [reply]
It make connivence from application view but with webinar case it seems to not optimal to get all other participants's connection status. May be we can use `publish` and `subscribe` config flag.

PS: publish config flag is true if we want to share peer's senders to all other peer's who enable subscribe flag to true (https://github.com/8xFF/atm0s-media-sdk-ts/blob/0d9fc1339efdd2495c4adfa51dfb619ff48d6516/packages/sdk-core/src/session.ts#L25-L26)

**Reply by jibon57** [reply]
Yes, that can be a good way. Or share connection info when someone share video tracks (webcam, screen share) so that other user can get idea about connection. But having dedicated option as you suggested will better.
